### ISSUE TITLE
Provider version 4.0.1 cannot plan CosmosDB resources from 3.x.x

### ISSUE THREAD

**Issue by cailyoung** [hashicorp/terraform-provider-azurerm#27300]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.5

### AzureRM Provider Version

4.0.1

### Affected Resource(s)/Data Source(s)

azurerm_cosmosdb_account, azurerm_cosmosdb_sql_container

### Terraform Configuration Files

```hcl
resource "azurerm_cosmosdb_sql_database" "db" {
  name                = "cosmos-sql-db-${var.environment}"
  resource_group_name = azurerm_cosmosdb_account.db-account.resource_group_name
  account_name        = azurerm_cosmosdb_account.db-account.name
  throughput          = 400
  lifecycle {
    prevent_destroy = true
  }
}

resource "azurerm_cosmosdb_sql_container" "db-sql-container" {
  name                  = "db-sql-container-${var.environment}"
  resource_group_name   = azurerm_cosmosdb_account.db-account.resource_group_name
  account_name          = azurerm_cosmosdb_account.db-account.name
  database_name         = azurerm_cosmosdb_sql_database.db.name
  partition_key_paths   = ["/id"]
  partition_key_version = 1
  throughput            = 400
  lifecycle {
    prevent_destroy = true
  }
}
```

### Debug Output/Panic Output

```shell
Plan: 0 to add, 1 to change, 0 to destroy.
╷
│ Warning: Failed to decode resource from state
│
│ Error decoding "azurerm_cosmosdb_account.releasebot-db-account" from prior
│ state: unsupported attribute "connection_strings"
╵
╷
│ Warning: Failed to decode resource from state
│
│ Error decoding "azurerm_cosmosdb_sql_container.releasebot-db-sql-container"
│ from prior state: unsupported attribute "partition_key_path"
╵
```

### Expected Behaviour

State migration should happen if attributes are deprecated

### Actual Behaviour

`plan` fails because the old state contains attributes that have now been removed.

### Steps to Reproduce

Create resources with a v3.x provider, then upgrade provider to 4.0.1 and run `terraform plan`

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. Seems it's a Terraform Core issue since the existing deprecated property in the state file isn't automatically removed via running `tf refresh`.

**Comment by neil-yechenwei** [comment]
We can add the state upgrader to resolve this issue. Now the [PR fix](https://github.com/hashicorp/terraform-provider-azurerm/pull/27276) has been merged. I have verified that the issue has gone after azurerm provider is upgraded to v4.1.0. Please try this latest version. Thanks.

**Comment by cailyoung** [comment]
@neil-yechenwei Yes, it's working now in the affected pipeline. Thanks for the fix!
